app/docs/dev/policies/meshtrafficpermission.md

```diff
@@ -0,0 +1,290 @@
+---
+title: MeshTrafficPermission (beta)
+---
+
+{% warning %}
+This policy uses new policy matching algorithm and is in beta state,
+it should not be mixed with [TrafficPermission](../traffic-permissions.md).
+{% endwarning %}
+
+## TargetRef support matrix
+
+| TargetRef type    | top level | to  | from |
+| ----------------- | --------- | --- | ---- |
+| Mesh              | ✅        | ✅  | ❌   |
+| MeshSubset        | ✅        | ✅  | ❌   |
+| MeshService       | ✅        | ✅  | ❌   |
+| MeshServiceSubset | ✅        | ✅  | ❌   |
+| MeshGatewayRoute  | ✅        | ❌  | ❌   |
+
+If you don't understand this table you should read [matching docs](/docs/{{ page.version }}/policies/matching).
+
+## Configuration
+
+### Action
+
+{{ site.title }} allows configuring one of 3 actions for a group of service's clients:
+
+* `ALLOW` - lets the requests through
+* `DENY` - blocks all the requests
+* `ALLOW_WITH_SHADOW_DENY` - lets the requests through but emits logs as if requests were denied
+
+## Examples
+
+### Service 'payments' allows requests from 'orders'
+
+{% tabs allow-orders useUrlFragment=false %}
+{% tab allow-orders Kubernetes %}
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshTrafficPermission
+metadata:
+  namespace: kuma-system
+  name: allow-orders
+  labels:
+    kuma.io/mesh: default
+spec:
+  targetRef: # 1
+    kind: MeshService
+    name: payments
+  from:
+    - targetRef: # 2
+        kind: MeshService
+        name: orders
+      default: # 3
+        action: ALLOW
+```
+
+Apply the configuration with `kubectl apply -f [..]`.
+
+{% endtab %}
+{% tab allow-orders Universal %}
+
+```yaml
+type: MeshTrafficPermission
+name: allow-orders
+mesh: default
+spec:
+  targetRef: # 1
+    kind: MeshService
+    name: payments
+  from:
+    - targetRef: # 2
+        kind: MeshService
+        name: orders
+      default: # 3
+        action: ALLOW
+```
+Apply the configuration with `kumactl apply -f [..]` or with the [HTTP API](../../reference/http-api).
+
+{% endtab %}
+{% endtabs %}
+
+#### Explanation
+
+1. Top level `targetRef` selects data plane proxies that implement `payments` service.
+    MeshTrafficPermission `allow-orders` will be configured on these proxies.
+
+    ```yaml
+    targetRef: # 1
+      kind: MeshService
+      name: payments
+    ```
+
+2. `TargetRef` inside the `from` array selects proxies that implement `order` service.
+    These proxies will be subjected to the action from `default.action`.
+
+    ```yaml
+    - targetRef: # 2
+        kind: MeshService
+        name: orders
+    ```
+
+3. The action is `ALLOW`. All requests from service `orders` will be allowed on service `payments`.
+
+    ```yaml
+    default: # 3
+      action: ALLOW
+    ```
+
+### Deny all
+
+{% tabs deny-all useUrlFragment=false %}
+{% tab deny-all Kubernetes %}
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshTrafficPermission
+metadata:
+  namespace: kuma-system
+  name: deny-all
+  labels:
+    kuma.io/mesh: default
```

Comment:
Let's remove the `default` mesh labels it's just overloading folks with more info than necessary